ow_plexil/src/plexil-adapter/FaultDependencies.cpp

```diff
@@ -0,0 +1,355 @@
+        if (i.second == "Procedure"){
+          m_procedures[i.first].inoperable = status;
+        }
+        // Otherwise we update subsystem and add it to stack
+        else{
+          if (visited.insert(i.first).second && m_subsystems.find(i.first) != m_subsystems.end()){
+            // increment or decrement non_local_fault flag
+            if (status){
+              m_subsystems[i.first].non_local_fault += 1;
+              m_subsystems[i.first].inoperable += 1;
+            }
+            else{
+              m_subsystems[i.first].non_local_fault -= 1;
+              m_subsystems[i.first].inoperable -= 1;
+            }
+            nodes.push(i.first);
+          }
+        }
+      }
+    }
+
+}
+/*void FaultDependencies::DebugPrint(){
+  // Prints out current fault info for all faults and subsystems
+  std::cout << "FAULTS: " << std::endl;
+  std::cout << "------------------------------------------------------------------------"
+            << "------------------------------------------------------------------------" 
+            << std::endl;
+  std::cout << std::setw(40) << std::left << "NAME" << std::setw(20) << "FAULTY" <<  std::endl;
+  std::cout << "------------------------------------------------------------------------"
+            << "------------------------------------------------------------------------" 
+            << std::endl;
+  for (auto fault: m_faults){
+    std::cout << std::left << std::setw(40) << fault.second.name << std::setw(20) << fault.second.faulty << std::endl;
+  }
+
+  std::cout << "\n" << std::endl;
+  std::cout << "SUBSYSTEMS/PROCEDURES STATUS" << std::endl;
+  std::cout << "------------------------------------------------------------------------"
+            << "------------------------------------------------------------------------" 
+            << std::endl;
+
+  std::cout << std::setw(25) << std::left << "NAME" << std::setw(25) << "TYPE" << std::setw(25) << "NON-LOCAL FAULT" 
+  << std::setw(25) << "LOCAL FAULT" << std::setw(25) << "FAULTY" << std::setw(25) << "INOPERABLE" << std::endl;
+  std::cout << "------------------------------------------------------------------------"
+            << "------------------------------------------------------------------------" 
+            << std::endl;
+  for (auto subsystem: m_subsystems){
+    std::cout << std::left << std::setw(25) << subsystem.second.name << std::setw(25) << "Subsystem" << std::setw(25) << 
+    subsystem.second.non_local_fault << std::setw(25) << subsystem.second.local_fault << std::setw(25) <<
+    subsystem.second.faulty << std::setw(25) << subsystem.second.inoperable << std::endl;
+  }
+  for (auto proc: m_procedures){
+    std::cout << std::left << std::setw(25) << proc.second.name << std::setw(25) << "Procedure" << std::setw(25) << 
+    "-" << std::setw(25) << "-" << std::setw(25) <<"-" << std::setw(25) << proc.second.inoperable << std::endl;
+  }
+  std::cout << std::endl;
+}
+*/
```

Comment:
Is this an alternate version of the function? Consider removing.

Reply:
Woops, sorry I was changing over to ROS logging and forgot to delete the old one.